bot/seasons/evergreen/bookmark.py

```diff
@@ -45,21 +50,75 @@ def __init__(self, bot: commands.Bot):
         embed.set_author(name=target_message.author, icon_url=target_message.author.avatar_url)
         embed.set_thumbnail(url=bookmark_icon_url)
 
+        error_embed = discord.Embed(
+            title=random.choice(ERROR_REPLIES),
+            description=f"Please enable DMs to receive the bookmark. "
+                        f"Once done you can retry by reacting with {Emojis.pin}",
+            colour=Colours.soft_red
+        )
+
         try:
             await ctx.author.send(embed=embed)
         except discord.Forbidden:
-            error_embed = discord.Embed(
-                title=random.choice(ERROR_REPLIES),
-                description=f"{ctx.author.mention}, please enable your DMs to receive the bookmark",
-                colour=Colours.soft_red
-            )
-            await ctx.send(embed=error_embed)
+            await ctx.send(embed=error_embed, delete_after=7.5)
+            sent_person = set()
         else:
-            log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'")
             await ctx.message.add_reaction(Emojis.envelope)
+            sent_person = {ctx.author}  # set of id who got the message
+
+        log.info(f"{ctx.author} bookmarked {target_message.jump_url} with title '{title}'.")
+        await ctx.message.add_reaction(Emojis.pin)
+
+        copied_embed = deepcopy(embed)
+
+        copied_embed.add_field(
+            name=f'Bookmarked from {ctx.author.name}.',
+            value=f'[Visit original message]({ctx.message.jump_url})',
+            inline=False
+        )
+
+        def check(reaction: discord.Reaction, user: discord.User) -> bool:
+            return (
+                user != self.bot.user
+                and reaction.emoji == Emojis.pin
+                and reaction.message == ctx.message
+                and user not in sent_person
+            )
+
+        while True:
+            try:
+                reaction, user = await self.bot.wait_for("reaction_add", timeout=60.0, check=check)
+            except asyncio.TimeoutError:
+                await ctx.message.clear_reactions()
+                return
+
+            log.info(f"{user.name} bookmarked {target_message.jump_url} with title '{title}' from '{ctx.author}'.")
```

Comment:
This should be moved much further down, so we only log when we successfully help someone bookmark the message.

A failing case example will be when we cannot send the DM to user, even after logging that "user bookmarked message". Either change the logging message or move this line further down.

Reply:
I was told in my first contribution to log even though we fail to send DM . In that case I should use `Tried to bookmark`.